tests/Unit/CliSamplingTest.php

```diff
@@ -175,4 +185,83 @@ public function test_it_applies_individual_sample_rates_to_scheduled_tasks(): vo
         $this->assertSame(100, $writes);
         $this->assertCount(0, $this->core->ingest->buffer);
     }
+
+    #[DataProvider('vendorCommands')]
+    public function test_it_samples_vendor_commands_separately(string $command): void
+    {
+        $ingest = $this->fakeIngest();
+        Artisan::command($command, fn () => 0);
+
+        $status = Artisan::handle($input = new StringInput($command), new NullOutput);
+        Artisan::terminate($input, $status);
+
+        $ingest->assertWrittenTimes(0);
+    }
+
+    #[DataProvider('vendorCommands')]
+    public function test_it_samples_vendor_commands_when_enabled(string $command): void
+    {
+        $ingest = $this->fakeIngest();
+        Artisan::command($command, fn () => 0);
+
+        Nightwatch::captureDefaultVendorCommands();
+
+        $status = Artisan::handle($input = new StringInput($command), new NullOutput);
+        Artisan::terminate($input, $status);
+
+        $ingest->assertWrittenTimes(1);
+    }
+
+    #[DataProvider('vendorCommands')]
+    public function test_it_samples_vendor_scheduled_tasks_separately(string $command): void
+    {
+        event(new CommandStarting('schedule:run', new StringInput(''), new NullOutput));
+
+        event(new ScheduledTaskStarting($this->app[Schedule::class]->command($command)->everyMinute()));
+
+        $this->assertFalse(Nightwatch::sampling());
+    }
+
+    #[DataProvider('vendorCommands')]
+    public function test_it_samples_vendor_scheduled_tasks_when_enabled(string $command): void
+    {
+        Nightwatch::captureDefaultVendorCommands();
+
+        event(new CommandStarting('schedule:run', new StringInput(''), new NullOutput));
+
+        event(new ScheduledTaskStarting($this->app[Schedule::class]->command($command)->everyMinute()));
+
+        $this->assertTrue(Nightwatch::sampling());
+    }
+
+    #[DataProvider('vendorCommands')]
+    public function test_it_samples_vendor_scheduled_tasks_when_explicitly_sampled(string $command): void
```

Comment:
I don't think we should sample vendor scheduled tasks unless the user calls `Nightwatch::captureDefaultVendorCommands()`, even if the sample rate is explicitly set via `Sample::rate($rate)`. What do you think?